appstudio-controller/controllers/appstudio.redhat.com/snapshotenvironmentbinding_controller.go

```diff
@@ -195,6 +211,51 @@ func (r *SnapshotEnvironmentBindingReconciler) Reconcile(ctx context.Context, re
 	return ctrl.Result{}, nil
 }
 
+func addComponentDeploymentCondition(binding *appstudioshared.SnapshotEnvironmentBinding, ctx context.Context, c client.Client, log logr.Logger) error {
+	if len(binding.Status.ComponentDeploymentConditions) > 1 {
+		// this should never happen, log and fix it
+		log.Info("snapshot environment binding has multiple component deployment conditions",
+			"binding name", binding.Name, "binding namespace", binding.Namespace)
+		binding.Status.ComponentDeploymentConditions = []metav1.Condition{}
+	}
+	if len(binding.Status.ComponentDeploymentConditions) == 0 {
+		binding.Status.ComponentDeploymentConditions = append(binding.Status.ComponentDeploymentConditions, metav1.Condition{})
+	}
+
+	total := len(binding.Status.GitOpsDeployments)
+	synced := 0
+	for _, deploymentStatus := range binding.Status.GitOpsDeployments {
+		deployment := apibackend.GitOpsDeployment{}
+		err := c.Get(ctx, types.NamespacedName{Name: deploymentStatus.GitOpsDeployment, Namespace: binding.Namespace}, &deployment)
+		if err != nil {
+			return err
+		}
+		if deployment.Status.Sync.Status == apibackend.SyncStatusCodeSynced {
+			synced++
+		}
+	}
+
+	ctype := ComponentDeploymentConditionAllComponentsDeployed
```

Comment:
Shouldn't it wait for all components to get in Sync before setting the Status= true and type = AllComponentsDeployed?

if synced == total {
    type: AllComponentsDeployed 
    status: True
    reason: CommitsSynced
    message: All components are deployed and in Sync.   
} else {
    - type: AllComponentsNotDeployed 
      status: False
      reason: CommitsUnsynced
      message: 1 of 3 components are deployed and in Sync.
}

Reply:
The type never changes, it's kind of like the name of the condition.  It's only the status that indicates what the condition is, so think of it as
  "AllComponentsDeployed" is "true"
or
  "AllComponentsDeployed" is "false"

The current logic is correct, it  will only set the status to true if all components are in sync, essentially in pseudocode
```
status = false
if all components are deployed
    status = true
```

Reply:
Yes I got it now, it makes sense.